diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..603e46d
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,42 @@
+# Generated Cmake file begin
+cmake_minimum_required(VERSION 3.1)
+
+project(rockspec2cmake C CXX)
+
+find_package(Lua REQUIRED)
+
+## INSTALL DEFAULTS (Relative to CMAKE_INSTALL_PREFIX)
+# Primary paths
+set(INSTALL_BIN bin CACHE PATH "Where to install binaries to.")
+set(INSTALL_LIB lib CACHE PATH "Where to install libraries to.")
+set(INSTALL_ETC etc CACHE PATH "Where to store configuration files")
+set(INSTALL_SHARE share CACHE PATH "Directory for shared data.")
+
+set(INSTALL_LMOD ${INSTALL_LIB}/lua/${LUA_VERSION_MAJOR}.${LUA_VERSION_MINOR} CACHE PATH "Directory to install Lua modules.")
+set(INSTALL_CMOD ${INSTALL_LIB}/lua/${LUA_VERSION_MAJOR}.${LUA_VERSION_MINOR} CACHE PATH "Directory to install Lua binary modules.")
+
+set(rockspec2cmake.init_SOURCES rockspec2cmake/init.lua)
+set(r2cmake_SOURCES r2cmake.lua)
+set(rockspec2cmake.CMakeBuilder_SOURCES rockspec2cmake/CMakeBuilder.lua)
+
+install(DIRECTORY ${BUILD_COPY_DIRECTORIES} DESTINATION ${INSTALL_SHARE}/rockspec2cmake)
+
+function(build_install KEYS DIR)
+    list(REMOVE_DUPLICATES KEYS)
+
+    foreach(KEY ${${KEYS}})
+        set(BASE_NAME ${KEYS}_${KEY})
+        install(FILES ${${BASE_NAME}_SRC} DESTINATION ${DIR}/${${BASE_NAME}_DST} RENAME ${${BASE_NAME}_RENAME})
+    endforeach(KEY)
+endfunction(build_install)
+
+build_install(BUILD_INSTALL_lua ${INSTALL_LMOD})
+build_install(BUILD_INSTALL_lib ${INSTALL_LIB})
+build_install(BUILD_INSTALL_conf ${INSTALL_ETC})
+build_install(BUILD_INSTALL_bin ${INSTALL_BIN})
+
+install(FILES ${rockspec2cmake.CMakeBuilder_SOURCES} DESTINATION ${INSTALL_LMOD}/rockspec2cmake RENAME CMakeBuilder.lua)
+install(FILES ${r2cmake_SOURCES} DESTINATION ${INSTALL_LMOD}/ RENAME r2cmake.lua)
+install(FILES ${rockspec2cmake.init_SOURCES} DESTINATION ${INSTALL_LMOD}/rockspec2cmake RENAME init.lua)
+
+
